scripts/generate-app-config/generatePagesConfig.js

```diff
@@ -0,0 +1,233 @@
+  }
+
+  return configCopy;
+};
+
+/**
+* Builds out config for a root page.
+*/
+const buildPageConfig = (filePaths, generatePagesOptions, namespace) => (
+  filePaths.reduce((acc, { filePath, entryPoint }) => {
+    // Break up the file path
+    const parsedPath = path.parse(filePath);
+    // Grab the type (doc, test, etc) and the name without the extension.
+    const { name, extension: fileType } = parseExtension(parsedPath.name);
+
+    let pages = acc[fileType];
+    if (!pages) {
+      pages = {};
+      acc[fileType] = pages;
+    }
+
+    const directory = parsedPath.dir;
+    // drop the period for the extenion
+    const ext = parsedPath.ext.slice(1);
+    // for jsx or js files, we want to drop the extention for including them, because we don't know if the file will be transpiled.
+    const fileName = (ext === 'jsx' || ext === 'js') ? parsedPath.name : parsedPath.base;
+    const contentPath = relativePath(path.join(directory, fileName));
+    const routes = getRoutes(directory, fileType, name, entryPoint);
+    const packageNamespace = getNamespace(directory, namespace);
+    // Name space all the generated config by pakcage.
+    const key = `${packageNamespace}:${routes[0]}`;
+
+    pages[key] = recurs(pages[key], routes, contentPath, ext, packageNamespace);
+    return acc;
+  }, {})
+);
+
+/**
+* Simple alpha sort. Copied from MDN, if I'm being honest.
+*/
+const alphaSort = (a, b) => {
+  const nameA = (a || '').toUpperCase(); // ignore upper and lowercase
+  const nameB = (b || '').toUpperCase(); // ignore upper and lowercase
+  if (nameA < nameB) {
+    return -1;
+  }
+  if (nameA > nameB) {
+    return 1;
+  }
+
+  // names must be equal
+  return 0;
+};
+
+/**
+* Sort first by group, then by alpha as a tie breaker.
+*/
+const sortPage = (a, b) => {
+  let result = alphaSort(a.group, b.group);
+  if (result === 0) {
+    result = alphaSort(a.name, b.name);
+  }
+  return result;
+};
+
+/**
+* Sort the pages objects and convert them into ordered arrays.
+*/
+const sortPageConfig = config => (
+  Object.values(config).sort(sortPage).map((page) => {
+    if (page.pages) {
+      // eslint-disable-next-line no-param-reassign
+      page.pages = sortPageConfig(page.pages);
+    }
+    return page;
+  })
+);
+
+/**
+* Generates the file representing page config, which is in turn consumed by route config.
+*/
+const generatePagesConfig = (siteConfig, production) => {
+  const { generatePages: generatePagesOptions, pagesConfig, navConfig, hotReloading } = siteConfig;
+  // If a pages config is supplied don't do this logic.
+  if (pagesConfig) {
+    return pagesConfig;
+  }
+
+  // Gather the types to search for.
+  const types = pageTypes(navConfig).join(',');
+
+  // Get the default search patterns for both normal and lerna mono repos.
+  const defaultPatterns = generatePagesOptions.searchPatterns.reduce((acc, { root, source, dist, entryPoint }) => {
+    const souceDir = (production || !hotReloading.enabled) ? dist : source;
```

Comment:
We need to add some type of error handling if dist or source is not provided correctly

Reply:
it will default to '' if dist or source are not provided.